### ISSUE TITLE
Add functionality to amend to a previous commit based on its hash

### ISSUE THREAD

**Issue by alfii** [alfii/.dotfiles#16]
Should quit if there are merge conflict.
Could also (maybe) be able to resolve merge conflicts if it's easy.

**Comment by EspenBrun** [comment]
Man vil nok ikke prøve å løse en merge conflict automatisk... Hvis ikke git er smart not til å ordne det så tror jeg ikke det er lurt å prøve å lage det.

**Comment by EspenBrun** [comment]
What exactly is the use case for this? Do you want to have both staged and unstaged files in the working area? And only the staged files should be added to an earlier commit?

**Comment by alfii** [comment]
sant, det blir nok fort komplisert å gjøre det selv

**Comment by EspenBrun** [comment]
Yep. But what about the use case? I think maybe the best thing would be for it to take everything in the working area and put that into the given commit, like how gcam alias takes everything. If not it does a lot of hidden stuff and it might be hard to figure out what it actually does.

Also, it would be a lot easier to implement. 

Instead of staging the files you would want to add to the earlier commit, you instead make a new commit of the stuff you do not want to add to the earlier commit.
